### PR TITLE
get case owners for SQL

### PR DESCRIPTION
**Pull request by snopoke**
While looking at this [Sentry error](https://sentry.io/dimagi/commcarehq/issues/225519132/) I noticed that the `set_cleanliness_flags_for_domain` method only works for Couch cases. I don't see any reason why that would be the case except for an oversight.

best reviewed per commit :fish:

### CODE REVIEW

File: corehq/apps/commtrack/tests/test_xml.py [hunk 1]
```diff
@@ -620,7 +618,7 @@ def test_archive_last_form(self):
         ledger_accessors = LedgerAccessors(self.domain.name)
         def _assert_initial_state():
             if should_use_sql_backend(self.domain):
-                self.assertEqual(3, LedgerTransaction.objects.filter(form_id=second_form_id).count())
+                self.assertEqual(3, LedgerTransaction.objects.using('default').filter(form_id=second_form_id).count())
```
Comment (dimagimon) [inline corehq/apps/commtrack/tests/test_xml.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/commtrack/tests/test_xml.py [hunk 2]
```diff
@@ -478,15 +478,14 @@ def test_blank_case_id_in_transfer(self):
...
-    @override_settings(ALLOW_FORM_PROCESSING_QUERIES=True)
     def test_device_report_submissions_ignored(self):
         """
         submit a device report with a stock block and make sure it doesn't
         get processed
         """
         def _assert_no_stock_transactions():
             if should_use_sql_backend(self.domain):
-                self.assertEqual(0, LedgerTransaction.objects.count())
+                self.assertEqual(0, LedgerTransaction.objects.using('default').count())
```
Comment (benrudolph) [inline corehq/apps/commtrack/tests/test_xml.py]: nice i like this change

**Review by benrudolph — APPROVED** [review]

**Comment by snopoke** [comment]
```
ERROR: corehq.form_processor.tests.test_form_dbaccessor:FormAccessorsTestsSQL.test_soft_delete
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/form_processor/tests/test_form_dbaccessor.py", line 363, in test_soft_delete
    form.delete(using='default')
  File "/mnt/commcare-hq-ro/corehq/sql_db/models.py", line 73, in delete
    return super(PartitionedModel, self).delete(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/base.py", line 958, in delete
    return collector.delete()
  File "/vendor/lib/python2.7/site-packages/django/db/models/deletion.py", line 288, in delete
    count = qs._raw_delete(using=self.using)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 623, in _raw_delete
    return sql.DeleteQuery(self.model).delete_qs(self, using)
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/subqueries.py", line 81, in delete_qs
    cursor = self.get_compiler(using).execute_sql(CURSOR)
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 835, in execute_sql
    cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/raven/contrib/django/client.py", line 114, in execute
    return real_execute(self, sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
ProgrammingError: relation "form_processor_xformattachmentsql" does not exist
LINE 1: DELETE FROM "form_processor_xformattachmentsql" WHERE "form_...
```

**Review by mkangia — APPROVED** [review]
Don't have much context on this. Looks good overall.

**Comment by snopoke** [comment]
:white_check_mark: finally

**Review by benrudolph — APPROVED** [review]

**Comment by snopoke** [comment]
@millerdev note that now all sharded models extend `PartitionedModel`

**Pull request merged**